src/syrupy/extensions/amber/serializer.py

```diff
@@ -95,6 +99,23 @@ def read_file(cls, filepath: str) -> "SnapshotFossil":
 
     @classmethod
     def serialize(
+        cls,
+        data: "SerializableData",
+        *,
+        exclude: Optional["PropertyFilter"] = None,
+        matcher: Optional["PropertyMatcher"] = None,
+    ) -> str:
+        """
+        After serializing, new line control characters are normalised. This is needed
+        for interoperablity of snapshot matching between systems that do not use the
+        same new line control characters. Example snapshots generated on windows os
+        should not break when running the tests on a unix based system and vice versa.
```

Comment:
Worth calling out as effectively snapshots will never show a control character mismatch.

Reply:
I'm okay with this, however it's a breaking change.

Reply:
Technically it should not be, in the sense that bumping to a new version will not break existing tests. Will add a test with single carriage returns to the before code to test this out

Reply:
Now I'm thinking since it's a breaking change how about we roll our own line end character for amber files; i.e. `\r` and `\r` get translated into ambr specific values and then terminated with ambr specific line end character; e.g.
```py
assert "a\rb\r\nc\nd" == snapshot
```
```ambr
  '
  a␍
  b␍␤
  c␤
  '
```

Reply:
We'd then switch to byte mote for reading/writing the snapshots?

Carriage return vs. Newline doesn't happen too frequently, and I find it does make it a bit harder to read the raw snapshot file that way (mainly due to the introduction of "noise")

Reply:
This would be a wider sweeping change though, than currently where only a few files need be updated